js/ssai_pulse.js

```diff
@@ -237,19 +273,17 @@ OO.Ads.manager(function(_, $)
       {
         adMode = true;
         this.currentAd = ad;
-        if (this.currentAd.ad)
-        {
-          this.currentAd.ad.id3AdId = this.currentId3Object.adId;
+        if (ad.ad && ad.ad.data && (ad.duration && typeof ad.duration === 'number')) {
+          this.adIdDictionary[ad.ad.data.id].curAdId = ad.id;
```

Comment:
@cacel @ipadilla4  we might want to also check if `ad.ad.data.id` exists

Reply:
@cacel @ipadilla4 is ad.ad.data.id guaranteed to exist here? I suspect it should, (especially if ad.ad.data exists) but we might want to add a null check just in case